app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,57 @@
+class ArticlesController < ApplicationController
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:index, :show]
+
+  respond_to :html, :json
+  responders :flash
+
+  def index
+    if params[:tag].present?
+      @articles = Article.tagged_with(params[:tag])
+    else
+      @articles = Article.all
+      respond_with(@articles)
+    end
```

Comment:
Good point! I'm glad you pointed this out because I decided to read about the responders gem, `respond_to`, and `respond_with` again to refresh my understanding of it. :smiley: